driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1362,16 +1374,17 @@ synchronized void init() {
 
             this.configuration.register(this);
 
-            this.executorQueue = new LinkedBlockingQueue<Runnable>();
-            this.executor = makeExecutor(NON_BLOCKING_EXECUTOR_SIZE, "worker", executorQueue);
-            this.blockingExecutorQueue = new LinkedBlockingQueue<Runnable>();
-            this.blockingExecutor = makeExecutor(2, "blocking-task-worker", blockingExecutorQueue);
-            this.reconnectionExecutor = new ScheduledThreadPoolExecutor(2, threadFactory("reconnection"));
-            // scheduledTasksExecutor is used to process C* notifications. So having it mono-threaded ensures notifications are
-            // applied in the order received.
-            this.scheduledTasksExecutor = new ScheduledThreadPoolExecutor(1, threadFactory("scheduled-task-worker"));
-
-            this.reaper = new ConnectionReaper(this);
+            ThreadingOptions threadingOptions = this.configuration.getThreadingOptions();
+            ThreadPoolExecutor tmpExecutor = threadingOptions.executor(clusterName);
+            this.executorQueue = tmpExecutor.getQueue();
+            this.executor = MoreExecutors.listeningDecorator(tmpExecutor);
```

Comment:
As discussed offline, I agree that we could use `*Service` interfaces, and `instanceof` checks and casts internally to access the queues.
I don't see any value in exposing the Guava wrappers. Implementations may choose to return them, when we call `MoreExecutors.listeningDecorator` internally it will be a no-op.